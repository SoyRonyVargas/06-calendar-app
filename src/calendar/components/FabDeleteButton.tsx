import useCalendarStore from '../hooks/useCalendarStore'
import { FabDeleteButtonWrapper } from '../styled'
import { FaTrash } from 'react-icons/fa'
import { Fragment } from 'react'

const FabDeleteButton = () => {

    const { handleDeleteEvent , hasSelectedEvent } = useCalendarStore()

    const handleDelete = () => {

        handleDeleteEvent()

    }

    return (
        <Fragment>
            {
                hasSelectedEvent
                &&
                <FabDeleteButtonWrapper>
                    <button onClick={handleDelete} className="button is-large is-danger">
                        <span className="icon">
                            <FaTrash />
                        </span>
                    </button>
                </FabDeleteButtonWrapper>
            }
        </Fragment>
    )
}

export default FabDeleteButton