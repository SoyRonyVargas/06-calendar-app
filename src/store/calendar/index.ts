export * from './calendarSlice'
export * from './selectors'
export * from './thunks'
