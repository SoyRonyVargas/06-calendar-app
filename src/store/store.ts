import { calendarSlice } from './calendar/calendarSlice'
import { configureStore } from '@reduxjs/toolkit'
import authSlice from './auth/authSlice'
import UISlice from './ui/uiSlice'

export const store = configureStore({
  reducer: {
    calendar: calendarSlice.reducer,
    auth: authSlice.reducer,
    ui: UISlice.reducer
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({ serializableCheck : false })
})

export type RootState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch