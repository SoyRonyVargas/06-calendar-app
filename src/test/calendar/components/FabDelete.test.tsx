import FabDeleteButton from "../../../calendar/components/FabDeleteButton"
import { render , screen } from '@testing-library/react'
import { Provider } from "react-redux"
import { store } from "../../../store"

describe('Probando el boton FabDelete', () => { 

    test('Deberia mostrarse correctamente', () => {
      
        const { container } = render(
            <Provider store={store}>
                <FabDeleteButton/>
            </Provider>
        )

        expect(container).toMatchSnapshot()

    })
        
})