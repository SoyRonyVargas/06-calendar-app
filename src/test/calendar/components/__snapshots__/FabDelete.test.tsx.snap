// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el boton FabDelete Deberia mostrarse correctamente 1`] = `<div />`;
